File: app/src/main/java/dev/blackcat/minauta/data/Account.java
package dev.blackcat.minauta.data;

import java.io.Serializable;

public class Account implements Serializable
{

    protected String username;
    protected String password;
    protected Session session;

    public Account()
    {}

    public String getUsername()
    {
        return username;
    }

    public void setUsername(String username)
    {
        this.username = username;
    }

    public String getPassword()
    {
        return password;
    }

    public void setPassword(String password)
    {
        this.password = password;
    }

    public Session getSession()
    {
        return session;
    }

    public void setSession(Session session)
    {
        this.session = session;
    }

}

File: app/src/main/java/dev/blackcat/minauta/data/Account.kt
package dev.blackcat.minauta.data

import java.io.Serializable

enum class AccountType {
    NATIONAL,
    INTERNATIONAL
}

data class StoredAccount(
    var username: String,
    var password: String,
    var accountType: AccountType,
    var selected: Boolean
) {
    constructor() : this("", "", AccountType.INTERNATIONAL, false)
}

data class Account(
        var username: String?,
        var password: String?,

        var sessionLimit: SessionLimit,
        var initialized: Boolean) : Serializable {

    constructor() : this("", "", SessionLimit(), false)

}
